services/cloudtrail_service.py

```diff
@@ -0,0 +1,50 @@
+import boto3
+import time
+from datetime import datetime, timedelta
+
+class CloudtrailService:
+    def __init__(self) -> None:
+        self.client = boto3.client("cloudtrail", region_name="eu-west-2")
+
+    def get_active_users(self):
+        username_key = "eventData.useridentity.principalid"
```

Comment:
Is this a little specific for a service named `CloudTrail`. Perhaps the method either accept the key as an argument OR the method name could be geared up to serve the dormant_user process only?

Reply:
Will make method name more specific